src/main/repositories/product_repository.py

```diff
@@ -0,0 +1,25 @@
+# This file has the product repository
+
+from src.main.repositories.generic_repository import GeneralRepository
+
+
+class ProductRepository(GeneralRepository):
+
+    def __init__(self):
+        self.products = {}
+
+    def add(self, product):
+        self.products[product.id] = product
```

Comment:
You should handle how you set id within your repository

Reply:
fixed